model_compression_toolkit/core/common/similarity_analyzer.py

```diff
@@ -41,74 +41,75 @@ def tensor_norm(x: np.ndarray, p: float = 2.0) -> np.float:
     """
     Compute the Lp-norm of a tensor x.
     Args:
-        x: Tensor to compute its norm
+        x: Flattened tensor to compute its norm
         p: P to use for the Lp norm computation
 
     Returns:
         Lp norm of x.
     """
-    return np.power(np.power(np.abs(x.flatten()), p).sum(), 1.0/p)
 
+    return np.power(np.power(np.abs(x), p).sum(axis=-1), 1.0/p)
 
-#########################
-# Similarity functions
-#########################
 
-def compute_mse(float_tensor: np.ndarray, fxp_tensor: np.ndarray, norm: bool = False, norm_eps: float = 1e-8) -> float:
+def flatten_tensor(t: np.ndarray, batch: bool) -> np.ndarray:
```

Comment:
Why do you need special function and batch flag? why not using np.ravel()?

Reply:
This is not a total flattening of the tensor, we flatten here all axis but the first one (batch) in case it is wanted (that is why the batch flag is for).
Correct me if I'm wrong, but np.ravel would return a 1-D array, wouldn't it? is there a way to use it for flattening all axis but the first one?

Reply:
Ok so no ravel because you want to get result per batch. But now tensor_norm return a tensor and not scalar.

Reply:
If it is preforming the operation per sample than it would return a tensor which is fine (a scalar per sample), otherwise it would return a scalar which is the norm of the entire batch.
I don't understand the problem

Reply:
If I want to get tensor norm (I don't have a batch dim, only simple tensor), I would expect to get a scalar right, because norm is scalar. In your new tensor_norm function I get a tensor (in previous one I got a scalar). Maybe batch flag which is off by default.